src/libstd/sync/mutex.rs

```diff
@@ -463,6 +463,20 @@ impl<T: ?Sized + fmt::Display> fmt::Display for MutexGuard<'_, T> {
     }
 }
 
+#[unstable(feature = "mutex_partial_eq", issue = "0")]
+impl<T: PartialEq> PartialEq for Mutex<T> {
+    fn eq(&self, other: &Self) -> bool {
+        if ptr::eq(&self, &other) {
+            true
```

Comment:
As mentioned [on the forum](https://users.rust-lang.org/t/is-it-possible-to-compare-the-same-unsafecell-with-itself/35626/4?u=cuviper), reflexive equality can only be assumed for `T: Eq`. You can see [`ArcEqIdent`](https://github.com/rust-lang/rust/blob/7de9402b77ded0d8ec9e1c554521b2121449ef2b/src/liballoc/sync.rs#L1750-L1754) and [`RcEqIdent`](https://github.com/rust-lang/rust/blob/7de9402b77ded0d8ec9e1c554521b2121449ef2b/src/liballoc/rc.rs#L1185-L1189) for the way to specialize this though.

Reply:
Would it work to simply compare the value to itself with
```rust
let reference = self.lock().unwrap();
reference.eq(& reference)
```

Reply:
I think so, relying on auto-`Deref` through the `MutexGuard` to the inner type. Or you could make that explicit with `*reference == *reference`. (Personally, I'd call that local `guard`.)